specification/walkthrough.md

```diff
@@ -0,0 +1,273 @@
+one-sample t-test can be performed to estimate population level effects. At each
+level of the analysis, we need to know which image inputs correspond to which
+design matrix, and more how to keep track of and combine outputs from the
+previous level at the current level of analysis.
+
+_BIDS Stats Models_ proposes a general machine-readable document to describe
+multi-stage neuroimaging analyses in a precise, yet flexible manner. We
+accomplish this by defining a _graph_ composed of **Nodes** representing each
+level of the analysis and **Edges** which define the flow of data from one Node
+to another. Within each {py:class}`~bsmschema.models.Node` we specify a
+{py:class}`~bsmschema.models.Model` to estimate, and at least one
+{py:class}`~bsmschema.models.Contrast` to define the computed outputs of each
+`Node`. Within each node we also specify how to group the incoming inputs into
+analysis units using the {py:attr}`~bsmschema.models.Node.GroupBy` directive.
+
+## A simple example
+
+In a [Simon task](https://openneuro.org/datasets/ds000101/versions/00004),
+participants were scanned for 2 runs and asked to indicate whether a diamond
+that was presented to the left or right of a central fixation cross was green or
+red. There were two conditions: color-spatial _congruent_ and _incongruent_
+trials.
+
+A basic analysis of this task is to determine which regions showed _greater
+activity for incongruent versus congruent trials_, across all participants.
+
+We can perform this analysis by first estimating a **run-level** timeseries
+model for "Incongruent" and "Congruent" trials--separately for each individual
+run. We then compute a contrast comparing _Incongruent > Congruent_ (IvC)
+trials. Next, we pass the resulting statistical maps for the contrast to a
+**subject-level** estimator, which computes the average _IvC_ effect for each
+subject separately. Finally, we pass the resulting estimates to a
+**dataset-level** estimator, which conduts a one-sample t-test across all of the
+subject estimates for the _IvC_ contrast.
+
+Let's visualize this model for 3 participants:
+
+![flow](images/flow_model.png)
+
+We can formally represent this analysis as **BIDS Stats Model**:
+
+```{literalinclude} examples/model-walkthrough_smdl.json
+:language: JSON
+```
+
+```{note}
+For this example, we have limited the model to three subjects using the `Input` key.
+```
+
+_BSM_ defines this multi-stage analysis as a Graph, with each level of analysis
+defined as a separate `Node` object.
+
+### Run-level Model
+
+First, we define a `Node` for the run level analysis.
+
+```json
+      "Level": "Run",
+      "Name": "run_level"
+```
+
+Note that the {py:attr}`~bsmschema.models.Node.Level` key is necessary for
+determining which input images are available to the estimator. At the `Run`
+level, there are many sources of possible variables, most notably `_events.tsv`
+files which define the timing of task-related events.
+
```

Comment:
```suggestion

We can now specify the model and the contrasts for that run level node:

```json
    {
      "Level": "Run",
      "Name": "run_level",
      "Model": {},
      "Contrasts": []
    }
```

```

Reply:
Why the empty model?